service/history/shard/context.go

```diff
@@ -680,10 +678,6 @@ Create_Loop:
 					)
 					s.closeShard()
 					break Create_Loop
-				} else {
-					// Shard is re-acquired successfully, and we know write to DB is guaranteed to
-					// complete, it's safe to update MaxReadLevel
-					s.updateMaxReadLevelLocked(transferMaxReadLevel)
```

Comment:
It's safe to remove this because if we hit this branch, `renewRangeLocked` succeeds and `transferMaxReadLevel ` is already updated inside `renewRangeLocked` function.